### PR TITLE
Allow nexpr version of float/int numpy conversion

### LINKED ISSUE #409: c++ error with conversion to integer
Using latest anaconda python, latest boost, latest pythran and
latest clang, the following code snippet causes a compile error:

import numpy

# pythran export bug(float[])
def bug(x):

        ints=numpy.zeros(5,'int')
        ints[0]=int(numpy.floor(x))

        return ints

#print bug(2.1)

The error is as follows:

%pythran bug.py

CRITICAL Leave the vessel! Women and children first!
C++ compiler failed to compile translated code.
E: Compile error!

******** Command line was: ********
'clang' '/tmp/tmpXBA5cx.cpp' '-I/opt/anaconda/include/python2.7' '-I/opt/anaconda/lib/python2.7/site-packages/numpy/core/include/numpy' '-I/opt/anaconda/lib/python2.7/site-packages/.' '-I/opt/anaconda/lib/python2.7/site-packages/pythran' '-I/opt/anaconda/lib/python2.7/site-packages/pythran/pythonic/patch' '-DENABLE_PYTHON_MODULE' '-DUSE_GMP' '-O2' '-g' '-std=c++11' '-O2' '-shared' '-o' 'bug.so' '-L/opt/anaconda/lib/python2.7/config' '-lpthread' '-ldl' '-lutil' '-lpython2.7' '-lboost_python' '-lgmp' '-lgmpxx' '-fPIC' '-ltcmalloc_minimal'

******** Output :  ********

In file included from /tmp/tmpXBA5cx.cpp:11:
/opt/anaconda/lib/python2.7/site-packages/pythran/pythonic/__builtin__/int_.hpp:13:24: error: no viable conversion from 'pythonic::types::numpy_expr<pythonic::numpy::proxy::floor, pythonic::types::ndarray<double, 1> >' to 'long'
                return t;
                       ^
/opt/anaconda/lib/python2.7/site-packages/pythran/pythonic/__builtin__/int_.hpp:25:38: note: in instantiation of function template specialization 'pythonic::__builtin__::int_<pythonic::types::numpy_expr<pythonic::numpy::proxy::floor, pythonic::types::ndarray<double, 1> > >' requested here
        PROXY(pythonic::__builtin__, int_);
                                     ^
/opt/anaconda/lib/python2.7/site-packages/pythran/pythonic/utils/proxy.hpp:21:35: note: expanded from macro 'PROXY'
#define PROXY(ns,f) FPROXY(ns, f, f)
                                  ^
/opt/anaconda/lib/python2.7/site-packages/pythran/pythonic/utils/proxy.hpp:15:28: note: expanded from macro 'FPROXY'
                return ns::f(std::forward<Types>(types)...); \
                           ^
/tmp/tmpXBA5cx.cpp:66:25: note: in instantiation of function template specialization 'pythonic::__builtin__::proxy::int_::operator()<pythonic::types::numpy_expr<pythonic::numpy::proxy::floor, pythonic::types::ndarray<double, 1> > >' requested here
    std::get<0>(ints) = pythonic::__builtin__::proxy::int_{}(pythonic::numpy::proxy::floor{}(x));
                        ^
/tmp/tmpXBA5cx.cpp:77:30: note: in instantiation of function template specialization '__pythran_bug::bug::operator()<pythonic::types::ndarray<double, 1> >' requested here
  return __pythran_bug::bug()(a0);
                             ^
1 error generated.

### PR DESCRIPTION
**Pull request by serge-sans-paille**
This fixes #409

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
@pbrunet this one is ready for merging!

**Comment by serge-sans-paille** [comment]
@pbrunet this one is ok to merge, if that's ok with you

File: pythran/pythonic/utils/meta.hpp [hunk 1]
```diff
@@ -14,6 +16,7 @@ template<bool V0> struct all_of<V0> {
 };
 
 template<bool... V> struct any_of;
+template<> struct any_of<> : std::integral_constant<bool, true> {};
```
Comment (pbrunet) [inline pythran/pythonic/utils/meta.hpp]: Sorry, I missed it during my first review. I think this value should be false and not true. Am I wrong?

Comment (serge-sans-paille) [inline pythran/pythonic/utils/meta.hpp]: On Wed, Mar 25, 2015 at 09:56:06AM -0700, pbrunet wrote:
> > @@ -14,6 +16,7 @@ template<bool V0> struct all_of<V0> {
> >  };
> >  
> >  template<bool... V> struct any_of;
> > +template<> struct any_of<> : std::integral_constant<bool, true> {};
> 
> Sorry, I missed it during my first review. I think this value should be false and not true. Am I wrong?

In math, any of an empty set is true...

Comment (pbrunet) [inline pythran/pythonic/utils/meta.hpp]: It is not the case in Python and in C++. It seem to be more "logic" to me as none of them is "true".

Comment (serge-sans-paille) [inline pythran/pythonic/utils/meta.hpp]: On Wed, Mar 25, 2015 at 10:37:33AM -0700, pbrunet wrote:
> > @@ -14,6 +16,7 @@ template<bool V0> struct all_of<V0> {
> >  };
> >  
> >  template<bool... V> struct any_of;
> > +template<> struct any_of<> : std::integral_constant<bool, true> {};
> 
> It is not the case in Python and in C++. It seem to be more "logic" to me as none of them is "true".

Yes, I am... 100% faulty on this one.

**Comment by serge-sans-paille** [comment]
travis is OK with this one \o/

**Pull request merged**